development/external-plugins/needs-tws/plugin.go

```diff
@@ -0,0 +1,305 @@
+		if err != nil {
+			return err
+		}
+		isBot := bc(e.Sender.Login)
+		if isBot {
+			// label removed by a bot. Skip.
+			return nil
+		}
+		deleted := e.Label.Name == lb
+		if !deleted {
+			// not a missing-docs label. Skip.
+			return nil
+		}
+		sender := e.Sender.Login
+		isCollaborator, err := p.ghc.IsCollaborator(org, repo, sender)
+		if err != nil {
+			return nil
+		}
+		if !isCollaborator {
+			// re-add label and notice only collaborators can bypass documentation review
+			err := p.ghc.AddLabel(org, repo, number, lb)
+			if err != nil {
+				l.Errorw("Failed to re-add label.", "error", err)
+				return err
+			}
+			return p.ghc.CreateComment(org, repo, number, fmt.Sprintf("@%s, you cannot bypass documentation review. Only team members with write access are allowed to do it.", sender))
+		}
+		return p.ghc.CreateComment(org, repo, number, fmt.Sprintf("Documentation review manually bypassed by @%s.", sender))
+	}
+	return nil
+}
+
+func (p PluginBackend) handlePullRequestReview(l *zap.SugaredLogger, re github.ReviewEvent) error {
+	if re.Action != github.ReviewActionSubmitted {
+		return nil
+	}
+	rc := reviewCtx{
+		author:      re.Review.User.Login,
+		issueAuthor: re.PullRequest.User.Login,
+		repo:        re.Repo,
+		body:        re.Review.Body,
+		htmlURL:     re.Repo.HTMLURL,
+		number:      re.PullRequest.Number,
+		assignees:   re.PullRequest.Assignees,
+	}
+	reviewState := github.ReviewState(strings.ToUpper(string(re.Review.State)))
+	if reviewState == github.ReviewStateApproved {
+		rc.approved = true
+	} else if reviewState == github.ReviewStateChangesRequested {
+		rc.approved = false
+	} else {
+		return nil
+	}
+	return p.handle(l, rc)
+}
+
+func (p PluginBackend) handle(l *zap.SugaredLogger, rc reviewCtx) error {
+	author := rc.author
+	issueAuthor := rc.issueAuthor
+	org := rc.repo.Owner.Login
+	repoName := rc.repo.Name
+	assignees := rc.assignees
+	number := rc.number
+	//body := rc.body
+	approved := rc.approved
+
+	isAuthor := author == issueAuthor
+	// author cannot review their own PR
+	if isAuthor {
+		return nil
+	}
+
+	gc, err := p.gcf.ClientFor(org, repoName)
+	if err != nil {
+		l.Errorw("Could not initialize git client.", "error", err)
+		return err
+	}
+	repoAliases, err := p.oac.LoadOwnersAliases(l, gc.Directory(), "OWNERS_ALIASES")
+	if err != nil {
+		l.Errorw("Could not fetch owners aliases.", "error", err)
+	}
+	twsGroup := repoAliases[DefaultTechnicalWritersGroup]
+	if !twsGroup.Has(author) {
+		l.Infof("'%s' is not in the group of required reviewers.", author)
+		return nil
+	}
+
+	var isAssignee bool
+	for _, a := range assignees {
+		if a.Login == author {
+			isAssignee = true
+			break
+		}
+	}
+
+	isCollaborator, err := p.ghc.IsCollaborator(org, repoName, author)
+	if err != nil {
+		l.Errorw("Could not determine if author is a collaborator.", "error", err)
+		return err
+	}
+
+	// User must be a collaborator
+	if !isCollaborator && !isAuthor {
+		return nil
```

Comment:
This should return some error with description. At present in logs we will get l.Errorw("Failed to handle Pull Request Review event", "error", err.Error()) where err.Error() will be simple a nil. That will be confusing when troubleshooting.

Reply:
This event should be ignore. If it's not meant to be ignored the function will return proper `err`, and will be seen on logs. Added explanation comment in the code.